Code/GraphMol/ChemReactions/DaylightParser.cpp

```diff
@@ -87,11 +87,39 @@ std::vector<std::string> splitSmartsIntoComponents(
   }
   return res;
 }
+
+ROMol* constructMolFromString(const std::string &txt,
+    std::map<std::string,std::string> *replacements, bool useSmiles, bool sanitize) {
+  ROMol *mol;
+  if(!useSmiles){
+    mol=SmartsToMol(txt,0,false,replacements);
+    if(sanitize){
```

Comment:
Last comment on this - if this is really only for depiction, why can't we just have a **PrepareForDepiction(ChemicalReaction &rxn)** function that always finds the rings  (and sanitization) for depiction purposes?  This would require no extra effort from the user and fix all depiction issues regardless of how the reaction was made.